protocol/synthetix/subgraph/mainnet/handleDelegationUpdated.ts

```diff
@@ -9,21 +14,43 @@ export function handleDelegationUpdated(event: DelegationUpdated): void {
     .concat('-')
     .concat(event.params.collateralType.toHex());
   let position = Position.load(id);
+
+  let vault = Vault.load(
+    event.params.poolId.toString().concat('-').concat(event.params.collateralType.toHex())
+  );
+  if (vault === null) {
+    vault = new Vault(
+      event.params.poolId.toString().concat('-').concat(event.params.collateralType.toHex())
+    );
+    vault.created_at = event.block.timestamp;
+    vault.created_at_block = event.block.number;
+    vault.collateral_amount = BigDecimal.fromString('0');
+    vault.collateral_type = event.params.collateralType.toHex();
+    vault.pool = event.params.poolId.toString();
+  }
+
+  let previous_position_amount = BigDecimal.fromString('0');
+  if (position) {
+    previous_position_amount = position.collateral_amount;
+    let amount_delta = event.params.amount.toBigDecimal().minus(previous_position_amount);
+    vault.collateral_amount = previous_position_amount.plus(amount_delta);
```

Comment:
```suggestion
    vault.collateral_amount = vault.collateral_amount.plus(amount_delta);
```
Vault is global and tracks all positions, not just one.